src/integrationTest/java/com/mongodb/spark/sql/connector/read/MongoBatchTest.java

```diff
@@ -419,6 +420,66 @@ void testReadsAreSupportedWithFilters() {
             .collectAsList());
   }
 
+  @Test
+  void testReadsAutomaticallyFilterRequiredNonExistentFields() {
```

Comment:
This integration test ensures only the expected number of documents are returned by by the connector.